compiler/sigmatch.nim

```diff
@@ -52,6 +52,8 @@ type
                              # this is used to prevent instantiations.
     genericConverter*: bool  # true if a generic converter needs to
                              # be instantiated
+    converterRetType*: PType # converter can turn let into var and var into let
+                             # used to check mutability differences          
```

Comment:
Can you alaborate, why you need a new field here and the existing fields are not sufficient?

Reply:
There is a comment there. When argument is converted the mutability check needs to be done not on the original argument, but on the return type of the converter hence return type needs to be made available to `matchesAux`.

It is likely possible to avoid this extra field if it is a problem. The cost of more code in `matchesAux` to unpack prepared call, skip nkHiddenDeref and nknkHiddenCallConv to find the return type of the converter.